src/main/java/org/broadinstitute/consent/http/models/DataAccessRequest.java

```diff
@@ -378,4 +385,11 @@ private Map<String, Object> shallowCopy(DataAccessRequest dar) {
     }
     return copy;
   }
+
+  // Simple state machine for determining if the Data Access Request
+  // is a progress report.
+  private void updateProgressReportState() {
+    progressReport = !getDraft() && (getParentId() != null);
+  }
```

Comment:
What if there is a DAR that is a draft and has a parent id?
I wonder if checking both of these fields will leave dars orphaned as neither a progress report nor a dar.

Reply:
Drafts cannot have progress reports.  It's a business rule enforced by src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java:251 when a progress report is submitted to the backend service.